nx/lib/nx/serving.ex

```diff
@@ -172,9 +172,41 @@ defmodule Nx.Serving do
          partitions: true}
       ]
 
-  If you have two GPUs, the serving will now gather batches and send
+  If you have two GPUs, `batched_run/3` will now gather batches and send
   them to the GPUs as they become available to process requests.
 
+  ### Distribution
+
+  All `Nx.Serving`s are distributed by default. If the current machine
+  does not have an instance of `Nx.Serving` running, `batched_run/3` will
+  automatically look for one in the cluster. The nodes do not need to run
+  the same code and applications. It is only required that they run the
+  same `Nx` version.
+
+  The load balancing between servings is done randomly, however, the number
+  of partitions are considered if the `partitioned: true` option is also given.
+  For example, if you have a node with 2 GPUs and another with 4, the latter
+  will receive the double of requests compared to the former.
+
+  `batched_run/3` receives an optional `distributed_preprocessing` callback as
+  third argument for preprocessing the input for distributed requests. When
+  using libraries like EXLA or Torchx, the tensor is often allocated in memory
+  inside a third-party library so it is necessary to either transfer or copy
+  the tensor to the binary backend before sending it to another node.
```

Comment:
If we support device-to-device transfer's already it would be more efficient to an expose an API for that in the future rather than depend on an additional copy. The NVIDIA implementation would use [NCCL](https://developer.nvidia.com/nccl) which would be more efficient than a copy+comms between nodes.

Not sure how this would best tie in though, and shouldn't block this going out anyway as this is already a pretty big performance win

Reply:
@seanmor5 in this case it may not matter much because we are not really planning to share the load between two gpus and the input data is relatively small (and many times it is not even a tensor yet!).

However, NCCL will play a role on training and pmap. Do you know how we tell XLA that we have a GPU on another _node_?

Reply:
There is a distributed PjRt client, so I'm guessing when nodes > 1 you need to make use of an API similar to this: https://github.com/tensorflow/tensorflow/blob/master/tensorflow/compiler/xla/pjrt/distributed/client.h